astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,343 @@
+
+        if instmode is not None:
+            link = link + "&instmode=" + instmode
+
+        if filter is not None:
+            link = link + "&filter=" + filter
+
+        if expflag is not None:
+            link = link + "&expflag=" + expflag
+
+        if expno is not None:
+            link = link + "&expno=" + expno
+
+        if name is not None:
+            link = link + "&name=" + name
+
+        if datasubsetno is not None:
+            link = link + "&datasubsetno=" + datasubsetno
+
+        if sourceno is not None:
+            link = link + "&sourceno=" + sourceno
+
+        if extension is not None:
+            link = link + "&extension=" + extension
+
+        response = self._handler.request('GET', link)
+        if response is not None:
+            response.raise_for_status()
+
+            if filename is None:
+                if "Content-Disposition" in response.headers.keys():
+                    filename = re.findall('filename="(.+)"',
+                                          response.headers[
+                                              "Content-Disposition"])[0]
+                else:
+                    filename = observation_id + ".tar"
+
+            self._handler.get_file(filename, response=response,
+                                   verbose=verbose)
+
+            if verbose:
+                log.info("Wrote {0} to {1}".format(link, filename))
+
+            return filename
+
+    def get_postcard(self, observation_id, image_type="OBS_EPIC",
+                     filename=None, verbose=False):
+        """
+        Download postcards from XSA
+
+        Parameters
+        ----------
+        observation_id : string
+            id of the observation for which download the postcard, mandatory
+            The identifier of the observation we want to retrieve, regardless
+            of whether it is simple or composite.
+        image_type : string
+            image type, optional, default 'OBS_EPIC'
+            The image_type to be returned. It can be: OBS_EPIC,
+            OBS_RGS_FLUXED, OBS_RGS_FLUXED_2, OBS_RGS_FLUXED_3, OBS_EPIC_MT,
+            OBS_RGS_FLUXED_MT, OBS_OM_V, OBS_OM_B, OBS_OM_U, OBS_OM_L,
+            OBS_OM_M, OBS_OM_S, OBS_OM_W
+        filename : string
+            file name to be used to store the postcard, optional, default None
+        verbose : bool
+            optional, default 'False'
+            Flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the observation postcard indicated
+        """
+
+        retri_type = "RETRIEVAL_TYPE=POSTCARD"
+        obs_id = "OBSERVATION_ID=" + observation_id
+        img_type = "OBS_IMAGE_TYPE=" + image_type
+        protocol = "PROTOCOL=HTTP"
+        link = self.data_url + "&".join([retri_type, obs_id,
+                                         img_type, protocol])
+
+        result = self._handler.request('GET', link, params=None)
```

Comment:
same comments as above - `link` should _just_ be the base URL, `params` should be the data you've encoded into the link in these lines

Reply:
Agree. Updated using params argument instead of the url parameters &.